Source/Core/Editor/Unity/PreBuildLinkXMLCreator.cs

```diff
@@ -0,0 +1,48 @@
+using System.IO;
+using UnityEditor.Build;
+using UnityEditor.Build.Reporting;
+
+/// <summary>
+/// Generates a link.xml file before the build process to preserve all Behaviors and Conditions from VR Builder add-ons.
+/// </summary>
+public class PreBuildLinkXMLCreator : IPreprocessBuildWithReport
+{
+  public int callbackOrder => 3;
+  public const string ResourceFolder = "Assets/MindPort/VR Builder/Resources";
+  public const string LinkXML = "link.xml";
+
+  public void OnPreprocessBuild(BuildReport report)
+  {
+    if (!Directory.Exists(ResourceFolder))
+    {
+      Directory.CreateDirectory(ResourceFolder);
+    }
+
+    string linkXmlPath = Path.Combine(ResourceFolder, LinkXML);
+    string linkXmlContent = @"<!-- Preserve all Behaviors and Conditions from VR Builder add-ons. -->
+<!-- The managed stripping (even set to low) will remove the add-on assembly if no entities of the add-on are used in the scene. -->
+<!-- This file is automatically generated by PreBuildLinkXMLCreator.cs and placed here when building. -->
+<linker>
+  <assembly fullname='VRBuilder.Animations' preserve='all' ignoreIfMissing='1'>
+    <namespace fullname='VRBuilder.Animations.Behaviors' preserve='all'/>
+  </assembly>
+  <assembly fullname='VRBuilder.Randomization' preserve='all' ignoreIfMissing='1'>
+    <namespace fullname='VRBuilder.Randomization.Behaviors' preserve='all'/>
+    <namespace fullname='VRBuilder.Randomization.Conditions' preserve='all'/>
+  </assembly>
+  <assembly fullname='VRBuilder.StatesAndData' preserve='all' ignoreIfMissing='1'>
+    <namespace fullname='VRBuilder.StatesAndData.Behaviors' preserve='all'/>
+    <namespace fullname='VRBuilder.StatesAndData.Conditions' preserve='all'/>
+  </assembly>
+  <assembly fullname='VRBuilder.TrackAndMeasure' preserve='all' ignoreIfMissing='1'>
+    <namespace fullname='VRBuilder.TrackAndMeasure.Behaviors' preserve='all'/>
+    <namespace fullname='VRBuilder.TrackAndMeasure.Conditions' preserve='all'/>
+  </assembly>
+  <assembly fullname='VRBuilder.Cognitive3D' preserve='all' ignoreIfMissing='1'>
+    <namespace fullname='VRBuilder.Cognitive3DIntegration.Behaviors' preserve='all'/>
+  </assembly>
+</linker>";
```

Comment:
I would also add VRBuilder.ProcessMenu, VRBuilder.Netcode, VRBuilder.Vivox and VRBuilder.Dissonance just to be sure

Reply:
Doing this will force the code to be included in builds even if they are not used.
I only focused on addons with Behaviors and Conditions as I don't see a reason why anything else should be striped.

Usually the biggest reason why stripping fails is when relaying on dependency injection runtime code.

Reply:
Ok, we can keep it like this and see if more issues arise - I agree it might be superfluous.